File: go/internal/replay/parserfixture/testdata/trees/javascript/orderService.js
const { OrderRepository } = require('./orderRepository');

class OrderService {
  constructor(repository) {
    this.repository = repository;
  }

  place(customerId, skus) {
    const order = this.repository.create(customerId);
    skus.forEach((sku) => order.addLine(sku));
    return this.repository.save(order);
  }

  find(orderId) {
    return this.repository.byId(orderId);
  }
}

module.exports = { OrderService };

File: go/internal/replay/parserfixture/testdata/trees/swift/OrderService.swift
import Foundation

final class OrderService {
    private let repository: OrderRepository

    init(repository: OrderRepository) {
        self.repository = repository
    }

    func place(customerId: String, skus: [String]) -> Order {
        var order = repository.create(customerId: customerId)
        for sku in skus {
            order.addLine(sku)
        }
        return repository.save(order)
    }

    func find(orderId: String) -> Order? {
        return repository.byId(orderId)
    }
}
